crates/bevy_ecs/src/system/commands/mod.rs

```diff
@@ -835,6 +836,46 @@ impl<'w, 's, 'a> EntityCommands<'w, 's, 'a> {
     pub fn commands(&mut self) -> &mut Commands<'w, 's> {
         self.commands
     }
+
+    /// Allows querying of an entity's components.
+    ///
+    /// # Panics
+    ///
+    /// The command will panic when applied if the associated entity and/or queried components do not exist.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # use bevy_ecs::prelude::*;
+    /// #
+    /// # #[derive(Resource)]
+    /// # struct PlayerEntity { entity: Entity }
+    /// #[derive(Component)]
+    /// struct Health(u32);
+    ///
+    /// fn zero_health(mut commands: Commands, player: Res<PlayerEntity>) {
+    ///     commands
+    ///         .entity(player.entity)
+    ///         .with_query::<&mut Health>(|mut health| {
+    ///             health.0 = 0;
+    ///         });
+    /// }
+    /// # bevy_ecs::system::assert_is_system(zero_health);
+    /// ```
+    pub fn with_query<Q>(
```

Comment:
Because Rust deprecated function generic defaults, I've added a second function, `with_query_filtered` which accepts the filter generic, and updated `with_query` to use `with_query_filtered` with a `()` for the filter type. I've replicated the examples and documentation for both methods, highlighting the ergonomic differences between them.